backend/de.metas.business/src/main/java/org/adempiere/mmovement/api/impl/MovementDAO.java

```diff
@@ -66,6 +68,17 @@ public List<I_M_Movement> retrieveMovementsForDDOrder(final int ddOrderId)
 				.list();
 	}
 
+	@Override
+	public BigDecimal retrieveMovementQtyForDDOrderLine(@NonNull final DDOrderLineId ddOrderLineId)
+	{
+		return queryBL.createQueryBuilder(I_M_MovementLine.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_M_MovementLine.COLUMNNAME_DD_OrderLine_ID, ddOrderLineId)
```

Comment:
please make sure there is an index on `M_MovementLine.DD_OrderLine_ID`. if there is none, I suggest you add one on `(DD_OrderLine_ID, MovementQty)`, so that this summing up is fast